ssawg_trending_scraper.py

```diff
@@ -43,112 +40,59 @@
 # ---------------------------------
-
-class Page():
+class BasePage():
     """
-    Establish page, URL, and HTML information.
+    Base class for establishing page, URL, and HTML information.
     """
+    auth = None
+
     def __init__(self, page):
         self.page = page
-    
-        def get_proper_urls_reports(page):
-            """
-            Get the correct URL for the quarterly timeframe; 50% through the quarter.
-            """
-            for quarter in range(4, 0, -1):
-                year = datetime.now().year
-                # creates the temporary url; starts at Quarter 4 and works backwards
-                url = f'{URL_ASPECT}/{self.page}/{year}/Q{quarter}/'
-                # this page requires a username/password
-                if self.page == "periscope_drift_reports":
-                    page_request = requests.get(url, auth=(USER_PERISCOPE, PASSWORD))
-                else:
-                    page_request = requests.get(url)
-                if page_request.status_code == 200:
-                    # use astropy Table 
-                    table_page = Table.read(page_request.text, format='ascii.html', htmldict={'table_id': 2})
-                    # pull quarterly start and stop dates from page
-                    start_time = table_page['TSTART'][0] 
-                    stop_time = table_page['TSTOP'][0]
-                    stop_minus_start = Chandra.Time.date2secs(stop_time) - Chandra.Time.date2secs(start_time)
-                    # define halfway through the quarter
-                    halfway = Chandra.Time.date2secs(start_time) + stop_minus_start
-                    # is now > 50% through quarter?
-                    if Chandra.Time.date2secs(datetime.now()) > halfway: 
-                        return f'{URL_ASPECT}/{self.page}/{year}/Q{quarter}/'
-                    # if not 50% through and it's the first quarter of the year
-                    elif quarter == 1:
-                        # switch to fourth quarter of previous year
-                        return f'{URL_ASPECT}/{self.page}/{year-1}/Q4/' 
-                    else:
-                        # try previous quarter
-                        return f'{URL_ASPECT}/{self.page}/{year}/Q{quarter-1}/' 
-                else:
-                    continue
-
-        def get_proper_urls_perigee(page):
-            """
-            Get the correct URL for the monthly perigee page; 50% through month.
-            """
-            now = datetime.now()
-            # if ~halfway through the month
-            if now.day > 15:
-                return f'{URL_ASPECT}/{self.page}/SUMMARY_DATA/{now.year}-M{now.month:02}/'
-            else:
-                last_month = datetime.now() + timedelta(days=-27)
-                return f'{URL_ASPECT}/{self.page}/SUMMARY_DATA/{last_month.year}-M{last_month.month:02}/'
-
-        # Generate the correct URLs for trending
-        if self.page in trending_pages["Reports"]:
```

Comment:
Here you violate encapsulation by having a method functionality depend in detail on the value of a non-trivial global constant.  That's where the sub-classing comes in, to define different behavior for different variants of the `Page` class.